### PR TITLE
core,netty,okhttp: make toString more consistent for channelz

### PR DESCRIPTION
**Pull request by zpencer**
Use MoreObjects.toStringHelper and use only the log id's long value,
because the class name is already present in the toStringHelper.

### CODE REVIEW

File: core/src/main/java/io/grpc/internal/InternalSubchannel.java [hunk 1]
```diff
@@ -390,6 +391,16 @@ public void shutdown(Status reason) {
...
 
+  @Override
+  public String toString() {
+    synchronized (lock) {
+      return MoreObjects.toStringHelper(this)
+          .add("logId", logId.getId())
+          .add("addressGroup", addressGroup.toString())
+          .toString();
+    }
+  }
```
Comment (dapengzhang0) [inline core/src/main/java/io/grpc/internal/InternalSubchannel.java]: nit: 

```java
Object addressGroupCopy;
synchronized (lock) {
  addressGroupCopy = addressGroup;
}
return MoreObjects.toStringHelper(this)
    .add("logId", logId.getId())
    ..add("addressGroup", addressGroupCopy)
    .toString();
```

File: netty/src/main/java/io/grpc/netty/NettyServer.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ public LogId getLogId() {
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
-        .add("logId", logId)
+        .add("logId", logId.toString())
```
Comment (dapengzhang0) [inline netty/src/main/java/io/grpc/netty/NettyServer.java]: why need `toString()`?

File: core/src/main/java/io/grpc/internal/InternalSubchannel.java [hunk 3]
```diff
@@ -390,6 +391,18 @@ public void shutdown(Status reason) {
     }
   }
 
+  @Override
+  public String toString() {
+    Object addressGroupCopy;
+    synchronized (lock) {
+      addressGroupCopy = addressGroup.toString();
```
Comment (dapengzhang0) [inline core/src/main/java/io/grpc/internal/InternalSubchannel.java]: I think you don't need call `toString()` here, EAG is immutable and pretty thread safe to call toString out of lock.

File: core/src/main/java/io/grpc/internal/InternalSubchannel.java [hunk 4]
```diff
@@ -390,6 +391,18 @@ public void shutdown(Status reason) {
...
+    }
+    return MoreObjects.toStringHelper(this)
+          .add("logId", logId.getId())
+          .add("addressGroup", addressGroupCopy.toString())
```
Comment (dapengzhang0) [inline core/src/main/java/io/grpc/internal/InternalSubchannel.java]: nit: no need to call `toString()` explicitly either, toStringHelper will do that for you.

File: core/src/main/java/io/grpc/internal/InternalSubchannel.java (hunk 3 above)
Comment (zpencer) [reply core/src/main/java/io/grpc/internal/InternalSubchannel.java]: The field may still be set in one thread but read in another thread, such as the server thread for Channelz.

Comment (dapengzhang0) [reply core/src/main/java/io/grpc/internal/InternalSubchannel.java]: >The field may still be set in one thread but read in another thread, such as the server thread for Channelz.

I don't see any problem with that. When you return the string, the returned value soon becomes stale even if you enclose everything of the toString method with the lock. toString() needn't be in the lock as long as the copying is in the lock
```java
synchronized (lock) {
   addressGroupCopy = addressGroup; 
   // no need to call toString() for a long list of addresses
}
```

Comment (zpencer) [reply core/src/main/java/io/grpc/internal/InternalSubchannel.java]: I see what you mean. Done.

**Pull request merged**